dtonator/src/main/java/com/bizo/dtonator/GenerateEnum.java

```diff
@@ -44,6 +52,29 @@ private void addEnumValues() {
     }
   }
 
+  private void addEnumGetters() {
+    try {
+      final Class<?> clazz = Class.forName(dto.getDomainType());
+      for (final Field field : clazz.getDeclaredFields()) {
+        if (!field.isEnumConstant() && !field.isSynthetic()) {
+          final GMethod getter = gc.getMethod("get{}", capitalize(field.getName())).returnType(field.getType());
+          getter.body.line("switch (this) {");
+          for (final String value : dto.getEnumValues()) {
+            @SuppressWarnings({ "unchecked", "rawtypes" })
+            final Enum enumClass = Enum.valueOf((Class<? extends Enum>) clazz, value);
+            final Object returnValue = PropertyUtils.getProperty(enumClass, field.getName());
+            getter.body.line("_ case {}: return {};", value, returnValue instanceof String ? "\"" + returnValue + "\"" : returnValue);
+          }
+          getter.body.line("}");
+          getter.body.line("return null;");
+        }
+      }
+    } catch (SecurityException | ClassNotFoundException | IllegalAccessException | InvocationTargetException | IllegalArgumentException
+      | NoSuchMethodException e) {
+      e.printStackTrace();
```

Comment:
If this is fatal, a throws would be better, if it's not fatal, but a log.warn/log.debug.